hazelcast/src/main/java/com/hazelcast/partition/impl/ReplicaSyncInfo.java

```diff
@@ -18,7 +18,7 @@
 
 import com.hazelcast.nio.Address;
 
-final class ReplicaSyncInfo {
+public final class ReplicaSyncInfo {
```

Comment:
Why does this need to be public? 

Nobody should access the internal of the com.hazelcast.partition.impl package apart from the node engine/ node since it needs to construct a InternalPartitionServiceImpl. 

The rest of the Hazelcast internals should talk to com.hazelcast.partition. 

Even for internal use, we try to prevent binding to the .impl package if a public API can be used.